src/Network/Wai/Middleware/Logging.hs

```diff
@@ -87,17 +92,23 @@ requestLogger = requestLoggerWith defaultConfig
 
 data Config = Config
   { cLogSource :: LogSource
+  , cGetClientIp :: Request -> Text
   , cGetDestinationIp :: Request -> Maybe Text
   }
 
 defaultConfig :: Config
 defaultConfig = Config
   { cLogSource = "requestLogger"
-  , cGetDestinationIp = fmap decodeUtf8 . lookupRequestHeader "x-real-ip"
+  , cGetClientIp = \req ->
+    fromMaybe (pack $ show $ remoteHost req)
+      $ (firstValue =<< lookupRequestHeader "x-forwarded-for" req)
```

Comment:
TIL about repeated headers.

Reply:
> repeated headers

:thinking: 

Are you suggesting that if I were to do,

```
curl \
  --header "X-Foo=bar" \
  --header "X-Foo=bat" \
  ...
```

The server would see `X-Foo=bar, bat`?

If so, then that's neat and TIL too!

I just assumed what was happening was akin to `curl --header "X-Foo=bar, bat" ...`. So it's just the client choosing to add meaning to an otherwise opaque, but single, header value.